### PR TITLE
uncheckedCast crusade

### PR DESCRIPTION
**Pull request by andr3ej**
Replace almost all uses of @ Suppress("UNCHECKED_CAST") with a function that is
* more specific as to exactly where you're cheating the type system
* enforces ancestral relationship between the 2 types
* usually less code

### CODE REVIEW

File: client/jackson/src/main/kotlin/net/corda/client/jackson/JacksonSupport.kt [hunk 1]
```diff
@@ -17,6 +17,7 @@ import net.corda.core.identity.AbstractParty
 import net.corda.core.identity.AnonymousParty
 import net.corda.core.identity.CordaX500Name
 import net.corda.core.identity.Party
+import net.corda.core.internal.uncheckedCast
```
Comment (mnesbit) [inline client/jackson/src/main/kotlin/net/corda/client/jackson/JacksonSupport.kt]: I do quite like this code as I agree the suppress is ugly. However, if we are trying to encourage use of this it shouldn't be in an internal utils, rather in the public utils.

Comment (mikehearn) [reply client/jackson/src/main/kotlin/net/corda/client/jackson/JacksonSupport.kt]: The thing is, such utilities might crop up in future releases of the standard library, and it'd be annoying if we were committed to a duplication forever.

File: client/jfx/src/main/kotlin/net/corda/client/jfx/utils/ObservableUtilities.kt [hunk 2]
```diff
@@ -120,13 +119,7 @@ fun <A> ObservableList<out A>.filter(predicate: ObservableValue<(A) -> Boolean>)
...
 fun <A> ObservableList<out A?>.filterNotNull(): ObservableList<A> {
     //TODO This is a tactical work round for an issue with SAM conversion (https://youtrack.jetbrains.com/issue/ALL-1552) so that the M10 explorer works.
-    @Suppress("UNCHECKED_CAST")
-    return (this as ObservableList<A?>).filtered(object : Predicate<A?> {
-        override fun test(t: A?): Boolean {
-            return t != null
-
-        }
-    }) as ObservableList<A>
+    return uncheckedCast(filtered { it != null })
```
Comment (mikehearn) [inline client/jfx/src/main/kotlin/net/corda/client/jfx/utils/ObservableUtilities.kt]: If the workaround isn't needed anymore please delete the comment too.

Comment (andr3ej) [reply client/jfx/src/main/kotlin/net/corda/client/jfx/utils/ObservableUtilities.kt]: I can't tell whether it's needed, so I've changed this to just add uncheckedCast with no other changes.

**Pull request merged**